File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool comp(string a, string b) { return a + b > b + a; }

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> arr;
    bool flag = true;
    for (int i = 0; i < numbers.size(); i++) {
        arr.push_back(to_string(numbers[i]));
        if (numbers[i] != 0)
                flag = false;
    }
    if (flag)
        return "0";
        
    sort(arr.begin(), arr.end(), comp);
    for (int i = 0; i < arr.size(); i++) answer += arr[i];
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] arr = new String[numbers.length];

        for (int i = 0; i < numbers.length; i++) {
            arr[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(arr, (i, j) -> Integer.compare(Integer.parseInt(j + i), Integer.parseInt(i + j)));
        
        for (String str : arr) {
            answer += str;
        }
    
        if (answer.charAt(0) == '0') return "0";
        
        return answer;
    }
}